interfaces/system_key.go

```diff
@@ -33,34 +38,68 @@ import (
 type systemKey struct {
 	BuildID          string   `yaml:"build-id"`
 	AppArmorFeatures []string `yaml:"apparmor-features"`
+	NFSHome          bool     `yaml:"nfs-home"`
+	Core             string   `yaml:"core,omitempty"`
 }
 
+var mockedSystemKey *systemKey
+
 func generateSystemKey() *systemKey {
-	var mySystemKey systemKey
+	// for testing only
+	if mockedSystemKey != nil {
+		return mockedSystemKey
+	}
 
-	// add build-id
+	var sk systemKey
 	buildID, err := osutil.MyBuildID()
 	if err != nil {
-		buildID = "unknown"
+		buildID = ""
 	}
-	mySystemKey.BuildID = buildID
+	sk.BuildID = buildID
+
+	// Add apparmor-feature (which is already sorted)
+	sk.AppArmorFeatures = release.AppArmorFeatures()
 
-	// Add apparmor-feature, note that ioutil.ReadDir() is already sorted.
+	// Add if home is using NFS, if so we need to have a different
+	// security profile and if this changes we need to change our
+	// profile.
+	sk.NFSHome, err = osutil.IsHomeUsingNFS()
+	if err != nil {
+		logger.Noticef("cannot determine nfs usage in generateSystemKey: %v", err)
+	}
+	// Add the current Core path, we need this because we call helpers
+	// like snap-confine from core that will need an updated profile
+	// if it changes
 	//
-	// We prefix the dirs.GlobalRootDir (which is usually "/") to make
-	// this testable.
-	mySystemKey.AppArmorFeatures = release.AppArmorFeatures()
+	// FIXME: what about core18? the snapd snap?
+	sk.Core, _ = os.Readlink(filepath.Join(dirs.SnapMountDir, "core/current"))
 
-	return &mySystemKey
+	return &sk
 }
 
 // SystemKey outputs a string that identifies what security profiles
 // environment this snapd is using. Security profiles that were generated
 // with a different Systemkey should be re-generated.
 func SystemKey() string {
-	sk, err := yaml.Marshal(generateSystemKey())
+	sk := generateSystemKey()
+
+	// special case: unknown build-ids always trigger a rebuild
+	if sk.BuildID == "" {
+		return ""
+	}
+	sks, err := yaml.Marshal(sk)
+	if err != nil {
+		panic(err)
```

Comment:
That looks a bit drastic even for an unlikely case like this. Should we return "" instead (forcing a rebuild as above) and log and error?

Reply:
while it's worth considering, note this is not new behaviour